corehq/apps/data_dictionary/tests/test_util.py

```diff
@@ -257,15 +257,48 @@ class UsedPropsByCaseTypeTest(TestCase):
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
-        case_block = CaseBlock(
+        cls.case_blocks = [
+            cls._create_case_block(
+                case_type='case-type',
+                name='Case A',
+                props={'prop': True},
+            ),
+            cls._create_case_block(
+                case_type='case-type',
+                name='Case B',
+                props={'other-prop': True},
+            ),
+            cls._create_case_block(
+                case_type='other-case-type',
+                name='Case C',
+                props={'prop': True},
+            ),
+            cls._create_case_block(
+                case_type='other-case-type',
+                name='Case A',
+                props={'foobar': True},
+            ),
+            cls._create_case_block(
+                case_type='test',
+                name='Case D',
+                props={},
+            ),
+        ]
+        case_search_es_setup(cls.domain, cls.case_blocks)
+
+    def _create_case_block(case_type, name, props):
+        return CaseBlock(
             case_id=uuid.uuid4().hex,
-            case_type='other-case-type',
-            case_name='Case A',
-            update={'other-prop': True},
+            case_type=case_type,
+            case_name=name,
+            update=props,
         )
-        case_search_es_setup(cls.domain, [case_block])
 
     def test_get_used_props_by_case_type(self):
         used_props_by_case_type = get_used_props_by_case_type(self.domain)
-        self.assertEqual(len(used_props_by_case_type), 1)
-        self.assertTrue('other-prop' in used_props_by_case_type['other-case-type'])
+        self.assertEqual(len(used_props_by_case_type), 3)
+        self.assertTrue({'prop', 'other-prop'}.issubset(used_props_by_case_type['case-type']))
+        self.assertFalse('foobar' in used_props_by_case_type['case-type'])
+        self.assertTrue({'prop', 'foobar'}.issubset(used_props_by_case_type['other-case-type']))
+        for prop in ['prop', 'other-prop', 'foobar']:
+            self.assertFalse(prop in used_props_by_case_type['test'])
```

Comment:
cant we directly compare the value in `used_props_by_case_type` with what we expect it to be?

Reply:
The reason I opted not to do this is that there is a bunch of other metadata properties, such as `@case_id` and `@owner_id`, which get added when creating a `CaseBlock` object. 

I couldn't find a const variable that lists what all the metadata variables are, so I thought it best to not create a static list in the test should these properties ever change in the future.

Reply:
Okay.

Feel free to punt this.

> The reason I opted not to do this is that there is a bunch of other metadata properties, such as @case_id and @owner_id, which get added when creating a CaseBlock object.

You could remove such metadata properties and test the rest that is relevant?

> should these properties ever change in the future

Right. I am also in a split when writing tests, thinking when should they not fail unnecessarily. I do err to side of caution and rather have them fail instead of being flexible. Plus, metadata is probably not relevant to the test here since we are confirming the case types and properties, though I don't see harm in retaining it.

Since we are deleting something based on this information, just being extra cautious that we test this more accurately though I might be overthinking.

Reply:
Hmm... good point about rather having tests more strictly fail than being too flexible in passing. If we create a CaseBlock with no props passed then we should have the complete list of metadata properties we can safely remove from the other tests. I'll make some adjustments to this test.

Reply:
[2c5022e](https://github.com/dimagi/commcare-hq/pull/33956/commits/2c5022eb581d0e13c7351d807a7c81a5ff839096)